lib/daru/core/group_by.rb

```diff
@@ -194,8 +194,15 @@ def get_group group
         indexes.each do |idx|
           rows << transpose[idx]
         end
+
+        new_index =
+          begin
+            @context.index[indexes]
+          rescue IndexError
+            indexes
+          end
         Daru::DataFrame.rows(
-          rows, index: @context.index[indexes], order: @context.vectors)
+          rows, index: new_index, order: @context.vectors)
       end
```

Comment:
Had no option but to do this because I made Index object return an error whenever invalid indexes are supplied. This really helps in keeping things simple. Moreover I think Index should have this responsibility to raise Exception when invalid indexes are supplied.